server/config_push_proxy.go

```diff
@@ -48,18 +54,44 @@ func FindConfigFile(fileName string) string {
 	return fileName
 }
 
-func LoadConfig(fileName string) {
-	fileName = FindConfigFile(fileName)
-	LogInfo("Loading " + fileName)
-
+// LoadConfig loads the config from the given file path.
+func LoadConfig(fileName string) (*ConfigPushProxy, error) {
 	file, err := os.Open(fileName)
 	if err != nil {
-		LogCritical("Error opening config file=" + fileName + ", err=" + err.Error())
+		return nil, err
+	}
+	defer file.Close()
+
+	buf, err := ioutil.ReadAll(file)
+	if err != nil {
+		return nil, err
 	}
 
-	decoder := json.NewDecoder(file)
-	err = decoder.Decode(CfgPP)
+	var cfg *ConfigPushProxy
+	err = json.Unmarshal(buf, &cfg)
 	if err != nil {
-		LogCritical("Error decoding config file=" + fileName + ", err=" + err.Error())
+		fmt.Println(buf, err)
+		return nil, err
+	}
+	// If both are disabled, that means an old config file is being used. Atleast enable console log.
+	if !cfg.EnableConsoleLog && !cfg.EnableFileLog {
+		cfg.EnableConsoleLog = true
 	}
+	if cfg.EnableFileLog {
+		if cfg.LogFileLocation == "" {
+			return nil, errors.New("log file location not specified")
+		}
+		// if file does not exist, create it.
+		if _, err := os.Stat(cfg.LogFileLocation); os.IsNotExist(err) {
```

Comment:
Does this handle relative and full paths?

Reply:
Yes.